Bugzilla/Version.pm

```diff
@@ -113,31 +113,78 @@ sub bug_count {
   return $self->{'bug_count'};
 }
 
+# Copied from Milestone::update()
 sub update {
   my $self = shift;
-  my ($changes, $old_self) = $self->SUPER::update(@_);
+  my $dbh  = Bugzilla->dbh;
+
+  $dbh->bz_start_transaction();
+  my $changes = $self->SUPER::update(@_);
 
   if (exists $changes->{value}) {
-    my $dbh = Bugzilla->dbh;
+
+    # The version value is stored in the bugs table instead of its ID.
     $dbh->do(
       'UPDATE bugs SET version = ?
                   WHERE version = ? AND product_id = ?', undef,
-      ($self->name, $old_self->name, $self->product_id)
+      ($self->name, $changes->{value}->[0], $self->product_id)
+    );
```

Comment:
I'm not sure it's correct to update bugs when the _default_ version changes.

I can see the reasoning for Milestone, as that's in the future, but the version field captures the version the bug is applicable to and it probably shouldn't be touched once it has been set by the reporter, even if it's the old default value.

Reply:
I was mistaken; this is when the _version_ is updated, not when the _default version_ is updated.